ovos_core/intent_services/commonqa_service.py

```diff
@@ -237,26 +243,25 @@ def _query_timeout(self, message: Message):
                 pass
 
             # invoke best match. `message` here already has source=skills ctx
-            self.speak(best['answer'], message.forward("", best))
+            if not best.get('handles_speech'):
+                self.speak(best['answer'], message.forward("", best))
             LOG.info('Handling with: ' + str(best['skill_id']))
-            cb = best.get('callback_data') or {}
+            response_data = {**best, "phrase": search_phrase}
             self.bus.emit(message.forward('question:action',
-                                          data={'skill_id': best['skill_id'],
-                                                'phrase': search_phrase,
-                                                'callback_data': cb}))
+                                          data=response_data))
             query.answered = True
         else:
             query.answered = False
         query.completed.set()
 
-    # TODO: Should `speak` be `_speak`?
     def speak(self, utterance: str, message: Message = None):
         """Speak a sentence.
 
         Args:
             utterance (str): response to be spoken
             message (Message): Message associated with selected response
         """
+        log_deprecation("Skills should handle `speak` calls.", "0.1.0")
```

Comment:
just needs bumping ovos-workshop, this speak method should be deleted and "answer" added to the response message so ovos-workshop can speak it

workshop is already using its backwards compat to account for this being done in core since it has been updated

Reply:
Related to the above discussion, this is called if `handles_speech` isn't present in the message data